sentinel/src/transitions/accept_patient_reports.js

```diff
@@ -175,10 +173,27 @@ const findValidRegistration = (doc, config, registrations) => {
 const addReportUUIDToRegistration = (doc, config, registrations, callback) => {
     const validRegistration = registrations.length && findValidRegistration(doc, config, registrations);
     if (validRegistration) {
-      return db.medic.put(validRegistration, callback);
+      return db.medic.put(validRegistration, (err) => {
+              if (err) {
+                if (err.name === 'conflict') {
+                  db.get(validRegistration._id, function(err, doc) {
+                    if (err) { 
+                      callback(err);
+                    }
+                    
+                    validRegistration._rev = doc._rev;
+                    db.medic.put(validRegistration, callback);
+                  });
+                } else {
+                  callback(err);
+                }
+              }
+
+              callback(null, true);
+            });
```

Comment:
I think we should add some unit tests for this code, to make sure the callbacks are called correctly and only once. If you're more comfortable with Promises, feel free to use them for this block.

Reply:
Also, when registrations are saved in the `_silenceRegistrations` block, the responses from CouchDB will include the new `rev`, which you can save and avoid having to re-request on conflict.

Reply:
👍